api/v2/storageproviders.go

```diff
@@ -125,37 +124,58 @@ func (s *apiV2) handleStorageProvidersSetInfo(c echo.Context, u *util.User) erro
 }
 
 type storageProviderResp struct {
-	Addr            address.Address `json:"addr"`
-	Name            string          `json:"name"`
-	Suspended       bool            `json:"suspended"`
-	SuspendedReason string          `json:"suspendedReason,omitempty"`
-	Version         string          `json:"version"`
+	Addr            address.Address       `json:"addr"`
+	Name            string                `json:"name"`
+	Suspended       bool                  `json:"suspended"`
+	SuspendedReason string                `json:"suspendedReason,omitempty"`
+	Version         string                `json:"version"`
+	ChainInfo       *miner.MinerChainInfo `json:"chain_info"`
 }
 
 // handleGetStorageProviders godoc
 // @Summary      Get all storage providers
-// @Description  This endpoint returns all storage providers
+// @Description  This endpoint returns all storage providers. Note: Value may be cached
 // @Tags         sp
 // @Produce      json
 // @Success      200  {object}  []storageProviderResp
 // @Failure      400           {object}  util.HttpError
 // @Failure      500           {object}  util.HttpError
 // @Router       /v2/storage-providers [get]
 func (s *apiV2) handleGetStorageProviders(c echo.Context) error {
+	ctx := context.TODO()
+
+	// Cache the Chain Lookup for this miner, looking up if it doesnt exist / is expired
+	key := util.CacheKey(c, nil)
+	cached, ok := s.extendedCacher.Get(key)
+	if ok {
+		out, ok := cached.([]storageProviderResp)
+		if ok {
```

Comment:
I think we might want to return an error if this is `!ok`, since the key was found but I can't be cast (why? something cheeky might be happening)